pumpkin-protocol/src/client/play/c_chunk_data.rs

```diff
@@ -55,23 +53,26 @@ impl<'a> ClientPacket for CChunkData<'a> {
                     data_buf.put_u8(block_size as u8);
                     // Palette length
                     data_buf.put_var_int(&VarInt(palette.len() as i32));
-                    let mut palette_map = HashMap::new();
-                    palette.iter().enumerate().for_each(|(i, id)| {
-                        palette_map.insert(*id, i);
+
+                    let mut palette_ids = Vec::with_capacity(palette.len());
```

Comment:
You could probably improve this by removing this Vec and using a lazy iterator instead. This vec seems to only be used for an iterator.

Reply:
Good idea, reusing the palette Vec works the same. No need to have the palette_ids Vec